compiler/circlechef/core/src/ModelChef.cpp

```diff
@@ -300,6 +306,34 @@ template <typename T> void cook_graph(const T &graph, CookParams &cp)
       // Create Data
       int32_t count = (element_count(dims) > 0) ? element_count(dims) : filler.arg_size();
       auto data_vec = chef->generate(count);
+      // pack for INT4 and replace data_vec
+      if (operand.type() == circlechef::TensorType::INT4)
+      {
+        uint32_t packed = (count + 1) / 2;
+        std::vector<uint8_t> data_packed(packed);
+        for (uint32_t idx = 0; idx < packed; ++idx)
+        {
+          uint32_t sidx = idx * 2;
+          data_packed[idx] = data_vec[sidx++] & 0x0f;
```

Comment:
1. It doesn't need to check if the given input range is valid?
2. Actually, I'm wondering if this packing really works. For instance, 8 and -8 represents 0b00001000 and 0b11111000 respectively. Then, if two -8 values are packed into one int8 value with this logic, it would become 0b10001000. It is same with the case where two 8 values are packed into one int8 value.

Reply:
Range check is added in `souschef` (#12761)